gradle/validation/rat-sources.gradle

```diff
@@ -0,0 +1,167 @@
+                // UMASS kstem
+                pattern(substring: "THIS SOFTWARE IS PROVIDED BY UNIVERSITY OF MASSACHUSETTS AND OTHER CONTRIBUTORS")
+                // Egothor
+                pattern(substring: "Egothor Software License version 1.00")
+                // JaSpell
+                pattern(substring: "Copyright (c) 2005 Bruno Martins")
+                // d3.js
+                pattern(substring: "THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS")
+                // highlight.js
+                pattern(substring: "THIS SOFTWARE IS PROVIDED BY THE REGENTS AND CONTRIBUTORS")
+            }
+
+            // MIT-like
+            substringMatcher(licenseFamilyCategory: "MIT  ", licenseFamilyName:"Modified BSD License") {
+                // ICU license
+                pattern(substring: "Permission is hereby granted, free of charge, to any person obtaining a copy")
+            }
+
+            // Apache
+            substringMatcher(licenseFamilyCategory: "AL   ", licenseFamilyName: "Apache") {
+                pattern(substring: "Licensed to the Apache Software Foundation (ASF) under")
+                // this is the old - school one under some files
+                pattern(substring: "Licensed under the Apache License, Version 2.0 (the &quot;License&quot;)")
+            }
+
+            substringMatcher(licenseFamilyCategory: "GEN  ", licenseFamilyName: "Generated") {
+                // <!-- svg files generated by gnuplot -->
+                pattern(substring: "Produced by GNUPLOT")
+                // <!-- snowball stemmers generated by snowball compiler -->
+                pattern(substring: "This file was generated automatically by the Snowball to Java compiler")
+                // <!-- parsers generated by antlr -->
+                pattern(substring: "ANTLR GENERATED CODE")
+            }
+
+            approvedLicense(familyName: "Apache")
+            approvedLicense(familyName: "The MIT License")
+            approvedLicense(familyName: "Modified BSD License")
+            approvedLicense(familyName: "Generated")
+        }
+    }
+
+    def printUnknownFiles() {
+        def ratXml = new XmlParser().parse(xmlReport)
+        def unknownLicenses = 0
+        ratXml.resource.each { resource ->
+            if (resource.'license-approval'.@name[0] == "false") {
+                // println('Unknown license: ' + resource.@name)
+                unknownLicenses++
+            }
+        }
+        if (unknownLicenses > 0) {
+            throw new GradleException("Found " + unknownLicenses + " files with " +
+                    "unknown licenses.")
+        }
+    }
+
+    @TaskAction
+    def rat() {
+        if (!reportDir.exists()) {
+            reportDir.mkdirs()
+        }
+        def origEncoding = System.getProperty("file.encoding")
+        try {
+            System.setProperty("file.encoding", "UTF-8") //affects the output of the ant rat task
```

Comment:
This is crappy. An XML file should be UTF-8 unless it declares a different encoding in the preamble. We should not have to adjust file.encoding (in fact, we must not) - it seems like a bug somewhere down the toolchain.

Reply:
Yea, I was a little surprised at it too, but didn't want to touch it when copying from the original implementation. I'm assuming they ran into some issue that we may or may not see. I'll comment it out for now, but leave a note so if we're in trouble later we might be able to identify why a bit easier.